These functions are used for automated quality filtering of functional gene high throughput sequencing (HTS) data 

The first function (ORF.finder) translates the nucleotide sequences to protein and separates the file into the sequences that contain at least one Open Reading Frame (ORF).
ORF.finder also separates the sequences into right and wrong length (protein)
Description of arguments:
[sequences]: your DNA sequence file in fasta format
[target]: the name of your target gene (optional; default is NULL)
[expected.length]: the expected length of the amino-acid sequences (optional; default is NULL)
[plus and minus]: used to search for a range of length rather than an exact length (plus/minus the expected length) (optional; default is 0 for both)
[ANTISENS]: search for ORF in the reverse DNA strand? (optional; default is FALSE)

The second function (check.BLASTp) is used to recover sequences that don't have the expected length but do match the expected protein on BLASTp
This function is ran only if the [expected.length] argument was not set to 'NULL' in the QC function
Description of arguments:
[AA.BLASTed]: the fasta file containing the protein sequences checked on BLASTp)
[searchFor]: terms to search in the BLAST results to recover sequences (e.g. ammonia monooxygenase, AmoA)
[To_exclude]: set of words used to exclude (optional; default is NULL)
[DNA.BLASTed]: the corresponding DNA sequences file if you want to recover the nucleotide sequences as well (optional; default is NULL)
[nhits]: specifies how many BLAST hits to check, default option is 1
[DIR]: used to specify the directory where the BLAST files are stored, default option is '.' i.e. files stored in the same folder as this R script

The third function (combine.fasta) will combine the fasta file generated at the first step (ORF of correct length) with the fasta file generated at the second step (recovered sequences after BLASTp)
Description of arguments:
[AA.sequences]: the amino-acid fasta file generated by the ORF.finder function
[DNA.sequences]: the original nucleotide fasta file (before ORF.finder)
[AA.recovered]: the the amino-acid fasta file generated by the check.BLASTp function (optional; default=NULL)
[performMSA]: as a final check, you can perform a multiple sequence alignment of your combined fasta file to identify outliers (optional; default is FALSE)
[SD]: if the [performMSA] argument was set to TRUE, use this argument to change the stringency for the identification of outliers (higher SD values= lower stringency) (defalut is SD=2)
(if [performMSA] argument was set to TRUE and this function returns a fasta file, check the sequences on BLASTp)

For training purposes, use the ASV.fa file and the BLASTp hit files provided
